app/src/main/kotlin/at/bitfire/davdroid/sync/Syncer.kt

```diff
@@ -62,62 +66,119 @@ abstract class Syncer {
     lateinit var context: Context
 
     @Inject
-    lateinit var db: AppDatabase
+    lateinit var collectionRepository: DavCollectionRepository
 
     @Inject
     lateinit var logger: Logger
 
+    @Inject
+    lateinit var serviceRepository: DavServiceRepository
+
+    abstract val authority: String
+    abstract val serviceType: String
+
+    val accountSettings by lazy { accountSettingsFactory.forAccount(account) }
+    val httpClient = lazy { HttpClient.Builder(context, accountSettings).build() }
+
+    private val dbCollections = mutableMapOf<HttpUrl, Collection>()
 
     /**
-     * Creates and/or deletes local collections (calendars, address books, etc) and updates them
-     * with remote information. Then syncs the actual entries (events, tasks, contacts, etc) of all
-     * collections.
+     * Creates, updates and/or deletes local resources (calendars, address books, etc) according to
+     * remote collection information. Then syncs the actual entries (events, tasks, contacts, etc)
+     * of the remaining up-to-date resources.
      */
-    abstract fun sync(account: Account, extras: Array<String>, authority: String, httpClient: Lazy<HttpClient>, provider: ContentProviderClient, syncResult: SyncResult)
-
-    fun onPerformSync(
-        account: Account,
-        extras: Array<String>,
-        authority: String,
-        syncResult: SyncResult
-    ) {
-        logger.log(Level.INFO, "$authority sync of $account initiated", extras.joinToString(", "))
+    fun sync(provider: ContentProviderClient) {
```

Comment:
Can we make it `private`? It really shouldn't be used by someone else than the `Syncer`.